src/units/attack_type.cpp

```diff
@@ -259,6 +259,10 @@ bool attack_type::matches_filter(const config& filter) const
 	return matches;
 }
 
+bool attack_type::is_active() const
+{
+	return self_ || other_ || self_loc_ != map_location::null_location() || other_loc_ != map_location::null_location();
```

Comment:
What does is_active mean? There's no documentation, and from the name I would expect this to check whether it's an "offense-only" attack, a "defense-only" attack, etc.